client/asset/eth/eth.go

```diff
@@ -450,8 +468,42 @@ func (eth *ExchangeWallet) Address() (string, error) {
 	return eth.acct.Address.String(), nil
 }
 
+// importAccount decrypts the seed extended private key, derives the private key used
+// for account of the wallet, and imports it into the eth node.
+func (eth *ExchangeWallet) importAccount(pw string) error {
+	seed, err := keygen.DecryptExtendedKey(eth.encSeedXPrivKey, []byte(pw))
+	if err != nil {
+		return err
+	}
+	child, err := keygen.GenDeepChild(seed, accountDerivationPath)
+	if err != nil {
+		return err
+	}
+	privKey, err := child.SerializedPrivKey()
+	if err != nil {
+		return err
+	}
+
+	acct, err := eth.node.importAccount(pw, privKey)
+	if !bytes.Equal(acct.Address.Bytes(), eth.acct.Address.Bytes()) {
+		return fmt.Errorf("account created by seed != account created by xPubKey")
+	}
+
+	return nil
+}
+
 // Unlock unlocks the exchange wallet.
 func (eth *ExchangeWallet) Unlock(pw string) error {
+	eth.importAccountMtx.Lock()
+	defer eth.importAccountMtx.Unlock()
+
+	if !eth.importedAccount {
+		err := eth.importAccount(pw)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I put this into unlock because that's the first time the wallet backend has access to the wallet password, which is needed to decrypt the private key that was passed during `Setup`. Although might not need to be here now that @buck54321 is adding some new methods on `asset.Driver`.